org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -70,10 +70,15 @@ class StateReader {
 }
 
 function convertDeviceDateToTranslatorDate(unixTimestamp) {
-    // Date takes a number of milliseconds, so convert seconds to milliseconds
-    var datetime = new Date(unixTimestamp * 1000);
+    // Only convert if input is valid numeric value
+    if (unixTimestamp && !isNaN(unixTimestamp)) { 
+        // Date takes a number of milliseconds, so convert seconds to milliseconds
+        var datetime = new Date(unixTimestamp * 1000);
+        return datetime.toISOString();
+    }
 
-    return datetime.toISOString();
+    // return all other invalid input as-is since we can't convert to DateTime
+    return unixTimestamp;
```

Comment:
>but our schema is likely derived off the WINK schema yes?

The same schema is used to define sensors coming from all providers, not just Wink.

>i see your point however looks like updated_at and changed_at might not always be the same - and if WINK distinguishes shouldnt we too?

The difference between the two is that one is reported by the physical device, and the other is when Wink sees the value change (which, I agree, could be different).

This is why I wanted to understand better the cases where Wink doesn't provide the _changed_at value. Their documentation says that this value is reported by the physical device. This suggests to me that device Arjun saw this with may not have reported a value yet. If that is by design for Wink, then we need a sane way to handle that case. 

_updated_at seemed like a good proxy for the _changed_at value, since they should be similar. If the scenario I outlined above is the only time this can happen, another alternative might be to use the device's [created_at](http://docs.winkapiv2.apiary.io/reference/device) timestamp, since, essentially the last time the value changed would be when the device was created in that scenario.

I don't think we should get in the habit of just returning values directly from the third parties in cases like this, since (unless we get lucky) what they're giving us will not match the format that our schema specifies. Our clients expect the data coming back from the translators to match the schema we say we translate to (otherwise what's the point of the schema?).

> but wouldnt that mean another query/lookup to the server for the property

No, the translator expects the entire Wink device payload. See [devicedata.json](https://github.com/openT2T/translators/blob/dev/org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/tests/devicedata.json) for an example of a Wink device payload (id fields were changed, and fields were added to show examples of every sensor value defined in [Wink's API](http://docs.winkapiv2.apiary.io/reference/device/sensor), but should otherwise be about what you should get. Note that a typical sensor would only have properties for the types of sensors included in that device, e.g. the GoControl motion sensor has the 'motion' field and the 'temperature' field, along with the motion_updated_at, etc fields. It wouldn't have the others).